src/components/FollowButton/index.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import './style.sass';
+import { isLoggedIn, api, getToken } from '../../utils';
+
+export class FollowBtn extends Component {
+  static propTypes = {
+
+  }
+
+  state = {
+    following: false,
+    loading: false,
+  }
+
+  componentWillMount() {
+    // fetch logged in user following
+    if (isLoggedIn()) {
+      const user = getToken();
+      const { user: { username } } = user;
```

Comment:
what do you want to achieve in this line?  seems to me that the `user` object has a property `user` that you are trying to rename it to `username`

Reply:
`getToken()` functions returns an object with this shape 
```
user: {
    username: 'map',
    token: 'token',
  }
```
In line 19, the code is destructing username from the user object above

Reply:
you can simply achieve that by 

`const {  username } = user`

Reply:
@Rodgers-M Tried it and its not working. You may not destruct username directly from the user object. 
```
var user = {
  user: {
      username: 'map',
      token: 'token',
   }
}

// wrong: outputes "direct distructuring username=>" undefined
let {username} = user;
console.log('direct distructuring username=>', username);

// right
let {user : {username}} = user;
console.log('right distructuring: username=>', username)
```

Reply:
@Rodgers-M We tried that and it got the user object so in order to get the username in the user, you have to destruct the user object where to you get the `username`, `email` and `token`.